tests/integration/test_saml.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
```

Comment:
This change is out of the scope of the PR. My suggestion is that, in the future, you open a different PR per change in order to:
* Make reviews easier
* Shorten the feedback cycle, and increasing delivery
* Easier git operations (comparisons/reverts/etc) across revisions

Reply:
Yes, this PR has ended up ballooning with a few issues/inconsistencies having been highlighted through the development process. Next time I will try split into smaller, more focussed PRs.